test/langtools/jdk/javadoc/doclet/testSnippetTag/TestSnippetMarkup.java

```diff
@@ -0,0 +1,577 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.Reader;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.io.UncheckedIOException;
+import java.io.Writer;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Function;
+import java.util.regex.MatchResult;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.NestingKind;
+import javax.tools.FileObject;
+import javax.tools.JavaFileObject;
+import javax.tools.StandardJavaFileManager;
+import javax.tools.ToolProvider;
+
+import builder.ClassBuilder;
+import toolbox.ToolBox;
+
+import static javax.tools.DocumentationTool.Location.DOCUMENTATION_OUTPUT;
+
+/*
+ * @test
+ * @bug 8266666
+ * @summary Implementation for snippets
+ * @library /tools/lib ../../lib
+ * @modules jdk.compiler/com.sun.tools.javac.api
+ *          jdk.compiler/com.sun.tools.javac.main
+ *          jdk.javadoc/jdk.javadoc.internal.tool
+ * @build javadoc.tester.* toolbox.ToolBox toolbox.ModuleBuilder builder.ClassBuilder
+ * @run main TestSnippetMarkup
+ */
```

Comment:
I was surprised to hear that this is a convention, so much so that I wrote a script to survey the JDK tests in this regard.
The script showed that while what you said is true for javadoc, it is not so true for other areas (even in langtools). That said, I can see the rationale for having this convention and will update the snippet tests accordingly.

Separately. If there's a value in mass-fixing the JDK tests so that they follow this convention, I can create a PR to do that. Here are some numbers provided by the script, to assess the situation. There are 32,920 `test/**/*.java` files. Of these, 19,190 have jtreg comments. Of these, 4,284 (or 22%) do not follow the convention one way or another. If pursued, that might result in an extremely tedious-to-review PR.

Reply:
Maybe the convention is not as strong as I thought. It definitely does not warrant mass updates.

I'd be OK to review any updates needed for javadoc tests.